lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -814,6 +814,11 @@ SPIRVType *LLVMToSPIRVBase::transSPIRVOpaqueType(StringRef STName,
 SPIRVType *LLVMToSPIRVBase::transScavengedType(Value *V) {
   if (auto *F = dyn_cast<Function>(V)) {
     FunctionType *FnTy = Scavenger->getFunctionType(F);
+    // VarArg functions other than printf are not supported in SPIR-V.
+    BM->getErrorLog().checkError(
+        !(FnTy->isVarArg() && F->getName() != "printf"),
```

Comment:
Is this test too strict?  In the test array_of_matrices.ll, I see:

declare dso_local spir_func i32 @_Z18__spirv_ocl_printfPU3AS2Kcz(ptr addrspace(2), ...)

Reply:
I agree, there is mangling, and at least 3 names for the printf, so `F->getName() != "printf"` would not work properly.

Reply:
I considered `F->getName().contains("printf")`, but thought that was too loose. What do you think, would that be better?

Reply:
In SPIR-V Backend we have something along the lines of
```
  if ((DemangledName.starts_with("__spirv_ocl_printf(") ||
       DemangledName.starts_with("printf(")) ...
```
so I guess as a first approach this could work after you've got `DemangledName` from `F->getName()` from `llvm::itaniumDemangle`.